### ISSUE TITLE
error-dispatching nethogs 

### ISSUE THREAD

**Issue by shirishag75** [raboof/nethogs#104]
![error-dispatching-nethogs](https://cloud.githubusercontent.com/assets/727626/20404926/f1244a4c-ad2c-11e6-9754-a9648986ffd1.png)

As can be seen it says error dispatching multiple times, have no idea why that error is there, as can be seen using the latest nethogs on debian stretch/testing.

**Comment by raboof** [comment]
Thanks for the report! This happens when an error occurs inside `pcap_dispatch`. It's interesting that there is 'normal' output for `eth0`.

My guess is that nethogs tries to monitor another device aside from `eth0`, and that that is not possible for some reason. I've added the device name and the more detailed error text to the error message. Could you try again with the latest master? It should still show an error message, but a more useful one :).

**Comment by shirishag75** [comment]
Compiled and running nethogs version 0.8.5-10-g5400ed3 whereas debian has 0.8.5-1 . Btw, I was looking and couldn't find Releases of nethogs on github.com. The benefit is that users and testers can at a glance know what are the newest features/unreleased releases have. See https://github.com/anrieff/libcpuid/releases as an example and the latest such as -

https://github.com/anrieff/libcpuid/compare/v0.3.0...master something similar would be much preferable.

**Comment by raboof** [comment]
You mean like https://github.com/raboof/nethogs/releases ? :)

**Comment by shirishag75** [comment]
oops....my bad. Didn't see that, yup, exactly like that :)

**Comment by shirishag75** [comment]
Just saw this -

![nethogs-somewhat-fixed](https://cloud.githubusercontent.com/assets/727626/20459008/c29e1564-aed9-11e6-9d4d-d2eaa2bbef61.png)

Would be much nicer if it had time-stamp in front of it and it came one below another, although don't know if it's a tall order :)

**Comment by raboof** [comment]
OK, at least's a clearer message :).

I suppose we could ignore this message and just assume if the interface is down it's not serving any traffic.

Presenting the errors more beautifully would be nice, but not a big priority right now - let's get them fixed instead :).

The 'problem' with bittorrent is that it tends to use many short-lived connections. What sometimes happens (as you can see) is that nethogs does see the traffic, but the connection is so short-lived that nethogs is too late to see it actually belongs to qtorrent. Of course in this case we could 'guess' based on the port number, but we'd rather not guess :). If you'd like to discuss matching such processes it'd be nice if you could open a separate issue.

**Comment by shirishag75** [comment]
agreed. Filed https://github.com/raboof/nethogs/issues/105 for the same. Hope its good.

**Comment by xd003** [comment]
Is there no way to fix this yet ? Would really appreciate it
